lib/influx.js

```diff
@@ -0,0 +1,38 @@
+import zlib from 'node:zlib';
+import { promisify } from 'node:util';
+
+/*
+* Sends compressed request metrics to InfluxDB
+* https://docs.influxdata.com/enterprise_influxdb/v1/guides/write_data/
+*/
+export const reportMetric = async (request, response, env) => {
+  const compress = promisify(zlib.gzip);
+
+  // Define API endpoint and headers
+  const url = `${env.INFLUX_URL}/api/v2/write?&bucket=${env.INFLUX_DATABASE}&precision=ms`;
+
+  return fetch(url, {
+    method: 'POST',
+    headers: {
+      'Authorization': `Token ${env.INFLUX_TOKEN}`,
+      'Content-Encoding': 'gzip',
+      'Content-Type': 'text/plain; charset=utf-8',
+    },
+    body: await compress(createMetricsFromRequest(request, response, env)),
```

Comment:
Nitpick: Adding gzip compression for one line that's 200-300 bytes long seems like overkill. The time saved on transmitting fewer bytes over the network may be less than the time needed to compress the data. I think the advice from InfluxDB docs applies to cases where the client writes a batch of metrics in one request.